doc/developer/wishlist.rst

```diff
@@ -0,0 +1,103 @@
+========
```

Comment:
This is a really nice list to have. Is it worth it to maybe have a tracking-issue on GitHub e.g. called "Wishlist" where this list is kept? It might be more natural to keep track of the list there rather than requiring a change to the documentation itself for any discussion/implementation of these features. You'd also get the nice cross-referencing between the closed issues "for free".